r-package/grf/R/forest_summary.R

```diff
@@ -115,69 +122,95 @@ test_calibration <- function(forest) {
 #' forest <- causal_forest(X, Y, W)
 #' best_linear_projection(forest, X[,1:2])
 #' }
-#' 
+#'
 #' @return An estimate of the best linear projection, along with coefficient standard errors.
 #'
 #' @importFrom stats lm
-#' 
+#'
 #' @export
-best_linear_projection <- function(forest, A = NULL, subset = NULL) {
-  
+best_linear_projection <- function(forest,
+                                   A = NULL,
+                                   subset = NULL,
+                                   num.trees.for.variance = 500,
+                                   seed = runif(1, 0, .Machine$integer.max)) {
+
   cluster.se <- length(forest$clusters) > 0
-  
+
   if (!("causal_forest" %in% class(forest))) {
     stop("`best_linear_projection` is only implemented for `causal_forest`")
   }
-  
+
   if (is.null(subset)) {
     subset <- 1:length(forest$Y.hat)
   }
-  
+
   if (class(subset) == "logical" & length(subset) == length(forest$Y.hat)) {
     subset <- which(subset)
   }
-  
+
   if (!all(subset %in% 1:length(forest$Y.hat))) {
     stop(paste(
       "If specified, subset must be a vector contained in 1:n,",
       "or a boolean vector of length n."
     ))
   }
-  
+
   clusters <- if (cluster.se) {
     forest$clusters
   } else {
     1:length(forest$Y.orig)
   }
   observation.weight <- observation_weights(forest)
-  
+
```

Comment:
We should add another test making sure clusters don't break this. I got an error when trying to do this on just one cluster out of two:

n <- 800
p <- 5
X <- matrix(rnorm(n * p), n, p)
W <- runif(n, 0, 1)
Y <- W * X[,1] + rnorm(n)
forest <- causal_forest(X, Y, W, clusters = c(rep(1, 400), rep(2, 400)))
best_linear_projection(forest, X[,1], which(forest$clusters == 1))

I think we can fix this by checking if forest$clusters has length > 1 instead of length > 0.